src/EditorFeatures/CSharpTest/ReverseForStatement/ReverseForStatementTests.cs

```diff
@@ -0,0 +1,368 @@
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (; i < args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutCondition()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; ; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutIncrementor()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; )
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutVariableReferencedInCondition()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; j < args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutVariableReferencedInIncrementor()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; j++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithoutVariableInitializer()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i; i < args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithMismatchedConditionAndIncrementor1()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; i--)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestMissingWithMismatchedConditionAndIncrementor2()
+        {
+            await TestMissingAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i >= args.Length; i++)
+        {
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveDeclarationNearReference)]
+        public async Task TestPostIncrement1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string[] args)
+    {
+        [||]for (int i = 0; i < args.Length; i++)
```

Comment:
Would it be worthwhile to have a test where a literal is used instead of `args.Length`? I'm guessing you'd gen `10 - 1` instead of `9`, but I'm not sure that `9` is preferable. Usually you want to mention the important number, not the number that is used as a practical detail useful to the implementation.

Reply:
tests added.